### DISCUSSION TITLE
A means of detecting when server is synced and ready

### DISCUSSION THREAD

**Discussion by catch-21** [CardanoSolutions/kupo#15] [Q&A] [answered]
#### Describe your idea, in simple words.
It would be good to have a field on the `health` endpoint to indicate when the server is ready to be queried.

#### Why is it a good idea?
Currently, the `health` endpoint can return `{"connection_status":"connected","most_recent_checkpoint":null,"most_recent_node_tip":null}` when it has established a connection with the node socket but it's sync status is unknown (reproducible for a short time when restoring from existing db).

The workaround is for client to wait for the absence of `null` values, and for checkpoint and tip to be equal before querying.

#### Are you willing to work on it yourself?
No

**Comment by KtorZ** [comment, marked as answer]
@james-iohk the server can be queries as soon as it is started. The question then becomes when is it relevant for a client to start consuming the data fetched from the server?

I don't think there's one generic answer for this as it very much depends on the use-case set by clients. Hence why the health endpoints gives both the current network tip and the latest checkpoint synced. It's then up to client to decide if they want to wait or not.

**Reply by caike** [reply]
Since "ready" depends on use-case, how do you feel about providing information on how long it took for `most_recent_checkpoint` to reach `most_recent_node_tip` since the start of the sync process ? Something like `seconds_to_node_tip` or `milliseconds_to_node_tip`

**Reply by KtorZ** [reply]
Please check the latest release 🥰

**Reply by caike** [reply]
Do you mean polling /metrics and checking `network_synchronization` until it reaches 1 ? Otherwise I'm not sure the new fields have the info I'm referring to, unless I'm missing something.
